modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/Form/WebtoolsAnalyticsRuleForm.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_webtools_analytics_rules\Form;
+
+use Drupal\Core\Entity\EntityForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class WebtoolsAnalyticsRuleForm.
+ */
+class WebtoolsAnalyticsRuleForm extends EntityForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+
+    /** @var \Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRuleInterface $rule */
+    $rule = $this->entity;
+
+    $form['id'] = [
+      '#type' => 'machine_name',
+      '#default_value' => $rule->id(),
+      '#machine_name' => [
+        'exists' => '\Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRule::load',
+        'source' => ['section'],
+      ],
+      '#disabled' => !$rule->isNew(),
+    ];
+
+    $form['section'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Section'),
```

Comment:
I am not sure on how this is supposed to work since there is no Behat test at the moment and the description doesn't explain it either. This is adding a free textfield for the site section, so the user can just enter whatever they want. Aren't these sections intended to be used in other parts of the module too? Should this maybe be an autocomplete or a select box?

Reply:
As far as I know the sections are arbitrary names that site managers can give to parts of their sites to make the tracking of visits more granular.